packages/server/src/modules/step/middleware.ts

```diff
@@ -0,0 +1,81 @@
+import { MiddlewareFunction } from '@types';
+import { Step } from '@chess-tent/models';
+import { StepNotFoundError, UnauthorizedStepEditError } from './errors';
+import * as service from './service';
+
+export const saveStep: MiddlewareFunction = (req, res, next) => {
+  service
+    .saveStep(res.locals.step as Step)
+    .then(next)
+    .catch(next);
+};
+
+export const publishStep: MiddlewareFunction = (req, res, next) => {
+  service.publishStep(res.locals.step.id).then(next).catch(next);
+};
+
+export const unpublishStep: MiddlewareFunction = (req, res, next) => {
+  service.unpublishStep(res.locals.step.id).then(next).catch(next);
+};
+
+export const patchStep: MiddlewareFunction = (req, res, next) => {
+  service.patchStep(res.locals.step.id, res.locals.step).then(next).catch(next);
+};
+
+export const getStep: MiddlewareFunction = (req, res, next) => {
+  service
+    .getStep(res.locals.step.id as Step['id'])
+    .then(step => {
+      if (!step) {
+        throw new StepNotFoundError();
+      }
+      res.locals.step = step;
+      next();
+    })
+    .catch(next);
+};
+
+export const deleteStep: MiddlewareFunction = (req, res, next) => {
+  service
+    .deleteStep(res.locals.step.id as Step['id'])
+    .then(next)
+    .catch(next);
+};
+
+export const findSteps: MiddlewareFunction = (req, res, next) => {
+  service
+    .findSteps(res.locals.filters)
+    .then(steps => {
+      if (!steps) {
+        throw new StepNotFoundError();
+      }
+      res.locals.steps = steps;
+      next();
+    })
+    .catch(next);
+};
+export const canEditStep: MiddlewareFunction = (req, res, next) => {
```

Comment:
I'd drop this one as well. We're not going to have steps without the new permission. Sine it's not going to be deployed anytime soon we can have routes unchecked.